S15/tests/benchmark.py

```diff
@@ -0,0 +1,312 @@
+            timeBehindMutex.release()
+        elif timeBehind[qPoolId] > 0:
+            timeBehindMutex.acquire()
+            timeBehind[qPoolId] -= loT
+            logging.info("QTYPE_%s recovering timeBehind %s, total is %s", qPoolId, loT, timeBehind[qPoolId])
+            timeBehindMutex.release()
+        else:
+            logging.debug('QTYPE_%s sleeping %s', qPoolId, loT)
+            time.sleep(loT)
+
+###############################################################################
+# Main. Starts all the threads. The threads will keep running for up to 24 h,
+# or until the program gets interrupted (e.g. with Ctrl-C). Logging goes to a
+# file in /tmp
+###############################################################################
+
+
+def main():
+
+    default_output_dir = os.path.join(
+        os.path.expanduser("~"), "runQueries_out")
+
+    parser = argparse.ArgumentParser(
+        description="Qserv Summer 15 Large Scale Test benchmark tool"
+        )
+
+    parser.add_argument('-v', '--verbose', dest='verbose', default=[],
+                        action='append_const',
+                        const=None,
+                        help='More verbose output, can use several times.')
+
+    parser.add_argument("-O", "--output-dir", dest="output_dir",
+                        default=default_output_dir,
+                        help="Absolute path to output directory. "
+                        "Default: %(default)s"
+                        )
+
+    parser.add_argument('master',
+                        default='ccqserv100.in2p3.fr',
+                        help='Qserv master/czar hostname'
+                        )
+
+    args = parser.parse_args()
+
+    try:
+        os.makedirs(args.output_dir)
+    except OSError:
+        if not os.path.isdir(args.output_dir):
+            raise
+
+    print("Output directory: {0}".format(args.output_dir))
+
+    verbosity = len(args.verbose)
+    levels = {0: logging.ERROR, 1: logging.WARNING, 2: logging.INFO,
+              3: logging.DEBUG}
+    simple_format = "[%(levelname)s] %(name)s: %(message)s"
+    level = levels.get(verbosity, logging.DEBUG)
+    logfile = os.path.join(args.output_dir, "runQueries.log")
+    logging.basicConfig(format=simple_format, level=level, filename=logfile)
+
+    query_pool_file = os.path.join(args.output_dir, "query_pools.log")
+    fout = open(query_pool_file, 'w')
+    pprint.pprint(QUERY_POOLS, fout, indent=2)
+    fout.close()
+
+    random.seed(123)
+
+    for queryPoolId in QUERY_POOLS:
+        qCount = CONCURRENCY[queryPoolId]
+        for i in range(0, qCount):
+            t = threading.Thread(target=runQueries, args=(queryPoolId,
+                                                          args.master,
+                                                          args.output_dir))
+            t.daemon = True
+            t.start()
+            t.join
+
+    time.sleep(60*60*48)
+
+if __name__ == "__main__":
+    main()
```

Comment:
There is a lot of commonality between `runQueries.py` and this file, should they be merged?